### ISSUE TITLE
Empty grid with multiselect and no header breaks if item is added

### ISSUE THREAD

**Issue by sevenzebras** [vaadin/framework#11607]
- Vaadin Framework version: 8.8.2
- Browser version: probably all (tested FF 60.7.0esr, IE11, Chrome 74.0.3729)
- Web container name and version: JBoss EAP 7.2
- Minimal reproducible example:

    	VerticalLayout lay = new VerticalLayout();

    	Grid<String> grid = new Grid<>();
        grid.setSelectionMode(SelectionMode.MULTI);
        grid.setHeaderVisible(false);
        grid.addColumn(Object::toString);

        List<String> items = new ArrayList<>();
        //items.add("ABC");
        grid.setItems(items);
        
    	lay.addComponent(grid);        
        lay.addComponent(new Button("add",e-> {
        	items.add("ABCDEFG");
        	grid.getDataProvider().refreshAll();
        }));
        setContent(lay);

- Expected behavior
After adding an item, the item is displayed as usual with a checkbox in front an no header above the grid.
- Actual behavior
![grafik](https://user-images.githubusercontent.com/14329634/58869432-7e287680-86be-11e9-963a-d7cfadd3ead3.png)
The issue is not present if there is at least one item in the grid during the initial rendering.

**Comment by TatuLund** [comment]
We can replicate your problem.

There is also slight problem in your code that causes problems with Grid. You cannot use String directly there. So I needed to change the code a bit.

There is also a workaround, if you call grid.recalculateColumnWidths(); after adding the item (possible after adding first item is enough) the problem disappears.

```
package com.example.myapplication;

import java.util.ArrayList;
import java.util.List;

import javax.servlet.annotation.WebServlet;

import com.vaadin.annotations.Theme;
import com.vaadin.annotations.VaadinServletConfiguration;
import com.vaadin.server.VaadinRequest;
import com.vaadin.server.VaadinServlet;
import com.vaadin.ui.Button;
import com.vaadin.ui.Grid;
import com.vaadin.ui.UI;
import com.vaadin.ui.VerticalLayout;

@Theme("mytheme")
public class MyUI extends UI {

	public class MyBean {
		private String name;

		public MyBean() {
			name = "ABCDEFG";
		}
		
		public String getName() {
			return name;
		}

		public void setName(String name) {
			this.name = name;
		}
		
	}
	
    @Override
    protected void init(VaadinRequest vaadinRequest) {
    	VerticalLayout lay = new VerticalLayout();

        Grid<MyBean> grid = new Grid<>();
        grid.setSelectionMode(Grid.SelectionMode.MULTI);
        grid.setHeaderVisible(false);
        List<MyBean> items = new ArrayList<>();
        grid.addColumn(item -> item.getName());

        // items.add("ABC");
        grid.setItems(items);

        lay.addComponent(grid);
        lay.addComponent(new Button("add", e -> {
            items.add(new MyBean());
            grid.getDataProvider().refreshAll();
            grid.recalculateColumnWidths();
        }));

        setContent(lay);
    }

    @WebServlet(urlPatterns = "/*", name = "MyUIServlet", asyncSupported = true)
    @VaadinServletConfiguration(ui = MyUI.class, productionMode = false)
    public static class MyUIServlet extends VaadinServlet {
    }
}

```

**Comment by ZheSun88** [comment]
Issue has been fixed in Master branch and it will be shipped with next maintenance version, 8.8.3 (estimated to release next week)
